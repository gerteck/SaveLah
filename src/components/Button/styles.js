import { StyleSheet } from 'react-native';
import { colors } from '../../utils/colors';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: '#4F63AC',
        width: '100%',
        alignSelf: 'center',
        textAlign: 'center',
        paddingVertical: 15,
        borderRadius: 8,
        marginTop: 30,
    },

    text: {
        textAlign: 'center',
        color: colors.white,
        fontSize: 16,
        fontWeight: 'bold',
    }
})
