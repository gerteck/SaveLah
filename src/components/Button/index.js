import React from "react";
import { TouchableOpacity, Text } from "react-native";
import { styles } from './styles';

const Button = ({title, onPress, style}) => {

    console.log("Inside Button Rerender")

    return (
        // hitSlop={20}
        <TouchableOpacity activeOpacity={0.6} onPress={onPress} 
            style={ [styles.container, style]}>
            <Text style={styles.text}>{title}</Text>
        </TouchableOpacity>
    
    )

}

export default React.memo(Button);