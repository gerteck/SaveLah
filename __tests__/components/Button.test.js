import React from "react";
import renderer from "react-test-renderer";
import Button from "../../src/components/Button";

describe("<Button />", () => {
    it('renders correctly across screens', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// run npm run test to run all .test.js files.