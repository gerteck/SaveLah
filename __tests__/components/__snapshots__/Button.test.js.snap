// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly across screens 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#4F63AC",
      "borderRadius": 8,
      "marginTop": 30,
      "opacity": 1,
      "paddingVertical": 15,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  />
</View>
`;
